File: 1485a_aad.cpp
/**
 * @file 1485a_aad.cpp
 * @author Ukeme Edet (ukemeedet2207@gmail.com)
 * @brief The solution to the problem Add and
 * Divide(https://codeforces.com/problemset/problem/1485/A)
 * @version 0.1
 * @date 2024-09-01
 *
 * @copyright Copyright (c) 2024
 *
 */

#include <bits/stdc++.h>

using namespace std;

/**
 * @brief The main function
 *
 * @return int
 */
int main(void) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    int t, a, b, res, i, q, ac;

    cin >> t;
    while (t--) {
        cin >> a >> b, res = INT_MAX, i = 0;

        b == 1 ? b++, i++ : 0;
        do {
            ac = a, q = 0;
            while (ac)
                ac /= b, q++;
            res = min(res, i + q);
            b++, i++;
        } while (i < res);
        cout << res << '\n';
    }
    return (EXIT_SUCCESS);
}

File: 1485a_aad.py
"""
@file: 1485a_aad.py
@author: Ukeme Edet (ukemeedet2207@gmail.com)
@brief: The solution to the problem Add and Divide(https://codeforces.com/contest/1485/problem/A)
@version: 1.0
@date: 2024-09-02

@copyright: Copyright (c) 2024
"""

from sys import stdin, stdout


def input():
    return stdin.readline().strip()


def print(string):
    return stdout.write(str(string) + "\n")


def main():
    t = int(input())
    for _ in range(t):
        a, b = [int(_) for _ in input().split()]
        res = 0
        ans = float("inf")
        if b == 1:
            res += 1
            b = 2
        while True:
            ac, c = a, 0
            while ac > 0:
                ac //= b
                c += 1
            ans = min(ans, res + c)
            res += 1
            b += 1
            if res > ans:
                break
        print(ans)


if __name__ == "__main__":
    main()
